generator/mavparse.py

```diff
@@ -248,16 +249,18 @@ def start_element(name, attrs):
             elif in_element == "mavlink.messages.message.extensions":
                 self.message[-1].extensions_start = len(self.message[-1].fields)
             elif in_element == "mavlink.messages.message.field":
-                check_attrs(attrs, ['name', 'type'], 'field')
-                print_format = attrs.get('print_format', None)
-                enum = attrs.get('enum', '')
-                display = attrs.get('display', '')
-                units = attrs.get('units', '')
-                if units:
-                    units = '[' + units + ']'
-                instance = attrs.get('instance', False)
-                new_field = MAVField(attrs['name'], attrs['type'], print_format, self, enum=enum, display=display, units=units, instance=instance)
                 if self.message[-1].extensions_start is None or self.allow_extensions:
+                    check_attrs(attrs, ['name', 'type'], 'field')
```

Comment:
This doesn't look right; did you mean to move all o fths code to within the "am I doing extensions now?" block?